src/ProjectSystem/Impl/Extensions/ProjectTreeExtensions.cs

```diff
@@ -85,5 +85,25 @@ public static class ProjectTreeExtensions {
             }
             return string.Empty;
         }
+
+        public static IEnumerable<string> GetAllFolderPaths(this IEnumerable<IProjectTree> nodes, UnconfiguredProject unconfiguredProject) {
+            List<string> paths = new List<string>();
```

Comment:
Other than `Directory.Exisits` I don't actually access any directories here. `Directory.Exisits`, returns `false` in case of any error while trying to determine if the directory exists. So, I don't think this is necessary.